app/services/submission/plant_trial_finalizer.rb

```diff
@@ -26,6 +26,12 @@ def create_new_trait_descriptors
     @new_trait_descriptors = (submission.content.step02.new_trait_descriptors || []).map do |attrs|
       attrs = attrs.with_indifferent_access
       attrs = attrs.merge(common_data)
+      if attrs[:trait].present?
+        attrs[:trait] = Trait.find_by_name(attrs['trait'])
+      end
+      if attrs[:plant_part].present?
```

Comment:
Not selecting a PP results in `attrs[:plant_part]` being equal to `""` which causes error on [creation of TD](https://github.com/eSpectrum-IT/brassica/pull/516/files#diff-40e0edd17f0d0745f6ef8476473e2c32R35) (just below).